rust-spec/lightclient/detection/detection.md

```diff
@@ -276,421 +447,352 @@ This implies that we need a timeout of at least *2 Delta* for remote
+            }
+            // At this point we do not care about the other error
+            // codes. We already have generated evidence for an
+            // attack and need to stop the lightclient. It does not
+            // help to call replace_primary. Also we will use the
+            // same primary to check with other secondaries in
+            // later iterations of the loop
+         }
+         // In the case where the secondary reports NoEvidence
+         // we do nothing
+    }
+    return Evidences; 
+}
+```
 
 - Expected precondition
-    - *FullNodes* is nonempty
+    - root and primary trace are a verification trace
 - Expected postcondition
-    - addr is moved from *Secondaries* to *FaultyNodes*
-    - an address *a* is moved from *FullNodes* to *Secondaries*
+    - solves the problem statement (if attack found, then evidence is reported)
 - Error condition
-    - if precondition is violated
+    - `ErrorTrustExpired`: fails if root expires (outside trusting
+    period) [[LCV-INV-TP.1]](LCV-INV-TP1-link)
+    - `ErrorNoPeers`: if no peers are left to replace secondaries, and
+    no evidence was found before that happened
 
-### From the verifier
+----
 
 ```go
-func VerifyToTarget(primary PeerID, lightStore LightStore,
-                    targetHeight Height) (LightStore, Result)
+func CreateEvidenceForPeer(peer PeerID, root LightBlock, trace LightStore)
+                          (Evidence, LightStore, result) {
+
+    common := root;
+
+    for i in 1 .. len(trace) {
+        auxLS, result := VerifyToTarget(peer, common, trace[i].Header.Height)
+  
+        if result != ResultSuccess {
+            // something went wrong; peer did not provide a verifyable block
+            return (nil, nil, nil, FaultyPeer)
+        }
+        else {
+            if auxLS.LatestVerified().Header != trace[i].Header {
+                // the header reported by the peer differs from the
+                // reference header in trace but both could be
+                // verified from common in one step.
+                // we can create evidence for submission to the secondary
+                ev := new InternalEvidence;
+                ev.Evidence.ConflictingBlock := trace[i];
+                ev.Evidence.CommonHeight := common.Height;
+                ev.Peer := peer
+                return (ev, common, auxLS, FoundEvidence)
+            }
+            else {
+                // the peer agrees with the trace, we move common forward
+                // we could delete auxLS as it will be overwritten in
+                // the next iteration
+                common := trace[i].Header
+            }
+        }
+    }
+    return (nil, nil, nil, NoEvidence)
+}
 ```
+- Expected precondition
+    - root and trace are a verification trace
+- Expected postcondition
+    - finds evidence where trace and peer diverge
+- Error condition
+    - `ErrorTrustExpired`: fails if root expires (outside trusting
```

Comment:
Should we return error obtain from VerifyToTarget so this is true?

Reply:
I added a clarification